### DISCUSSION TITLE
Syncing with multiple clients

### DISCUSSION THREAD

**Discussion by tonydero** [Lurkars/gloomhavensecretariat#75] [Q&A] [answered]
Hello and thanks for creating and providing this excellent tool!

I'm self-hosting a server with docker and can access it without problem, and seem to be able to use it without problem. We haven't done a whole lot yet, but that's not really the point of this question.

My question is if multiple devices/clients connect to the host/server, why doesn't the same information show? For example, I go to the page on my phone and add a character. Then when I go to the same page on my PC, it doesn't show anything. It's like it's a separate instance for each client. Is this working as intended? I tried messing with adding public server connection settings, but it doesn't connect. Edit: I'm guessing this is where my issue lies, based on the help pages. I'm not sure what to put here for the docker instance server settings.

I fully admit that I may be completely misunderstanding how this tool works. We've not even played GH once yet, we just don't want to use the stickers on our first play and this seemed like a great option.

Apologies if there is an obvious answer here. Many thanks for any assistance!

**Comment by Lurkars** [comment, marked as answer]
Hello, I am a bit unsure about your setup.

So to connect to the same server with the same game, each client needs to enter the server data and same password under "Connect to Server". You can select the public server and just choose a password as game identify. On every client enter the same pw and game state should be synced. On a self hosted server, it must be first accessible via network (local or exposed publicly). Usual case would be just your homenetwork, so your server machine has an IP like 192.168.1.34 for example. Then on every client you enter the IP as host, default port to 8080 and choose the same password.

https://help.gloomhaven-secretary.de/#server

**Reply by tonydero** [reply]
Apologies for being unclear!

I am locally running the server via the `docker-compose.yaml` provided in the GitHub repo (aside from changing the locally mapped port to `8525:80`. I've attempted to put in the server's IP (tried the server's actual IP, `localhost`, `127.0.0.1`, and `0.0.0.0`) and port (tried both `8525` and just `80`) and it won't connect. I am connecting to the self-hosted page and putting in the server info on that page's "Connect to Server" options. I'm guessing this is some sort of issue with the internal docker network maybe? Just hoping someone else might have run into this and figured it out.

**Reply by Lurkars** [reply]
Hey, the are a few things here:
- there is no IP 0.0.0.0, you need the local IP of your machine (usual something like 192.168.1.123, starting with 192.168.)
- the server is not running on port 80, but 8080, so port must be 8080:8080, or if you want to change outgoing port to 8525 to 8525:8080
- then is the question, which client are you using, if client runs under HTTPS, server must also has SSL enabled, because every modern browser rejects unencrypted connections then, for that case it's best when you also running client yourself without SSL (easier then running server with SSL, depending on setup) for that, server can also serve client or you using docker compose file with both, server and client, I just added a docker compose file for that now, see https://github.com/Lurkars/gloomhavensecretariat/blob/main/docker-compose.server.yaml so if you run this and using client running on port 80 to connect to server running on port 8080, it should work. As mentioned, the IP stuff you still need to figure out, on the hosting machine, localhost should work, but your other clients need the local IP of the server

**Reply by tonydero** [reply]
That was it! I was mistakenly thinking the server and client were both served at the same address/port (and from the same image). Thanks for bearing with me.
